### PR TITLE
For #1514: Better follow app bar specs

### PR DESCRIPTION
**Pull request by sblatz**
### Before

Screen Shot 2019-07-01 at 10 38 45 AM

### After
Screen Shot 2019-07-01 at 10 43 27 AM

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt [hunk 1]
```diff
@@ -356,13 +362,12 @@ internal class DisplayToolbar(
         val navigationActionsWidth = navigationActions
             .asSequence()
             .mapNotNull { it.view }
-            .fold(0) { usedWidth, view ->
-                val viewLeft = usedWidth
+            .fold(0) { _, view ->
```
Comment (NotWoods) [inline components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt]: This loop looks like it's supposed to sum together the width of all the actions, so the `usedWidth` argument shouldn't be discarded.

**Comment by sblatz** [comment]
I was able to update most of the tests to pass, but I'm struggling with getting these tests working:

* url box position is enclosing icon, url and page actions
* url box position is enclosing icon, url and page actions

There's a mismatch in the widths 😕

File: components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt (hunk 1 above)
Comment (sblatz) [reply components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt]: Ah, my bad. Fixed!

**Comment by sblatz** [comment]
@pocmo let me know if you want to pair on this at all :)

**Comment by pocmo** [comment]
Yeah, that may be a good idea. :)

A quick update: I did test this a bit today and it seems like the resizing does break some of the custom touch backgrounds (since the buttons do not seem to be a square anymore) used by Firefox for Echo Show.

**Comment by pocmo** [comment]
Closing for now. Let's get back to that in one of the next sprints. There are a bunch of toolbar improvements that we could group together and address in one sprint.

**Pull request closed without merging**